test/performance/io/format_fastq_benchmark.cpp

```diff
@@ -0,0 +1,242 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#include <benchmark/benchmark.h>
+
+#include <cstring>
+#include <seqan3/std/filesystem>
+#include <fstream>
+
+#include <seqan3/alphabet/nucleotide/dna5.hpp>
+#include <seqan3/alphabet/quality/phred42.hpp>
+#include <seqan3/core/detail/debug_stream_type.hpp>
+#include <seqan3/io/sequence_file/input.hpp>
+#include <seqan3/io/sequence_file/output.hpp>
+#include <seqan3/range/views/to_char.hpp>
+#include <seqan3/test/performance/sequence_generator.hpp>
+#include <seqan3/test/seqan2.hpp>
+#include <seqan3/test/tmp_filename.hpp>
+
+#if SEQAN3_HAS_SEQAN2
+#include <seqan/seq_io.h>
+#endif // SEQAN3_HAS_SEQAN2
+
+constexpr unsigned default_seed = 1234u;
+
+// ============================================================================
+// generate fastq file
+// ============================================================================
+
+constexpr size_t default_sequence_length = 50; //length of nucleotide and quality sequence
+
+std::string generate_fastq_file(size_t const entries_size)
+{
+    std::ostringstream stream_buffer{};
+    seqan3::debug_stream_type format_stream{stream_buffer};
+
+    std::string const id{"@name"};
+
+    auto seed = default_seed;
+
+    for (size_t i = 0; i < entries_size; ++i, ++seed)
+    {
+        auto random_sequence = seqan3::test::generate_sequence<seqan3::dna5>(default_sequence_length, 0, seed);
+        auto random_qualties = seqan3::test::generate_sequence<seqan3::phred42>(default_sequence_length, 0, seed);
```

Comment:
As you already use `default_sequence_length` I suggest deleting line 43 and use
```suggestion
        auto random_sequence = seqan3::test::generate_sequence<seqan3::dna5>(default_sequence_length, 0, default_seed);
        auto random_qualities = seqan3::test::generate_sequence<seqan3::phred42>(default_sequence_length, 0, default_seed);
```

Reply:
Ah the seed seems to be varied for every generated sequence.

Reply:
true ;)